agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2088,6 +2095,56 @@ bool backhaul_manager::get_sta_bssid(const sMacAddr &sta_mac, sMacAddr &bssid)
     return false;
 }
 
+bool backhaul_manager::handle_associated_sta_link_metrics_query(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                                const std::string &src_mac)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received ASSOCIATED_STA_LINK_METRICS_QUERY_MESSAGE , mid=" << std::dec
+               << int(mid);
+
+    if (!cmdu_tx.create(mid,
+                        ieee1905_1::eMessageType::ASSOCIATED_STA_LINK_METRICS_RESPONSE_MESSAGE)) {
+        LOG(ERROR)
+            << "cmdu creation of type ASSOCIATED_STA_LINK_METRICS_RESPONSE_MESSAGE, has failed";
+        return false;
+    }
+
+    auto mac = cmdu_rx.getClass<wfa_map::tlvStaMacAddressType>();
+    if (!mac) {
+        LOG(ERROR) << "Failed to get mac address";
+        return false;
+    }
+
+    auto assoc_link_metrics = cmdu_tx.addClass<wfa_map::tlvAssociatedStaLinkMetrics>();
+    if (!assoc_link_metrics) {
+        LOG(ERROR) << "Failed to create tlvAssociatedStaLinkMetrics tlv";
+        return false;
+    }
+
+    // Check if it is an error scenario - if the STA specified in the STA link Query message is not associated
+    // with any of the BSS operated by the Multi-AP Agent
+    sMacAddr bssid;
+    if (!get_sta_bssid(mac->sta_mac(), bssid)) {
+        LOG(ERROR) << "client with mac address " << mac->sta_mac() << " not found";
+        //Add an Error Code TLV
+        auto error_code_tlv = cmdu_tx.addClass<wfa_map::tlvErrorCode>();
+        if (!error_code_tlv) {
+            LOG(ERROR) << "addClass wfa_map::tlvErrorCode has failed";
+            return false;
+        }
+        error_code_tlv->reason_code() =
+            wfa_map::tlvErrorCode::STA_NOT_ASSOCIATED_WITH_ANY_BSS_OPERATED_BY_THE_AGENT;
+        error_code_tlv->sta_mac() = mac->sta_mac();
+
+        LOG(DEBUG) << "Send a ASSOCIATED_STA_LINK_METRICS_RESPONSE_MESSAGE back to controller";
+        return send_cmdu_to_bus(cmdu_tx, src_mac, bridge_info.mac);
+    } else {
+        LOG(DEBUG) << "client with mac address " << mac->sta_mac() << " connected to " << bssid;
+        //TODO
+        return false;
+    }
+}
```

Comment:
```suggestion
    }
    
    LOG(DEBUG) << "client with mac address " << mac->sta_mac() << " connected to " << bssid;
    //TODO
    return false;
}
```
`//TODO` what?

Also, if you already found the `bssid` to which the STA is connected, it's better to forward the CMDU the correct slave from here. Otherwise (by returning `false`) the message will be forwarded to all the slaves.